Front end update for fake server

dudeyds opened the pull request:
Add Axios to create POST request to server

Commit:
fake login server

Commit:
Merge branch 'master' into fake-login-server

Commit:
Added npm start, updated readme

Commit:
checking password as well as username

Commit:
Updated README

dudeyds committed:
Front end update for interacting with server

Post reuest added using Axiom to get response from express server,

CORS  added to server via NPM to get rid of Cross origin reource blocking error. Reccomended to configure CORS to increase security

dudeyds committed:
Merge branch 'master' into Front-end-update-for-fake-server

dudeyds committed:
Update .gitignore

Woops, extra carriage return

dudeyds committed:
Update .gitignore

Newline removed

goofballLogic commented on script.js:
```diff
@@ -39,13 +39,26 @@ class App extends React.Component {
-        password } });
-
-
+		axios.post('http://localhost:3000',{
```
this does not appear to target the correct endpoint. When I try it I get a 404.

dudeyds replied:
Endpoint corrected, and after I said I would wait for the server's endpoint to change before pulling this, very clever of me

goofballLogic commented on script.js:
```diff
@@ -39,13 +39,26 @@ class App extends React.Component {
+			password: password
+		})
+		.then((response) => {
+			console.log(response);
```
Probably shouldn't leave console logs in code for master branch

dudeyds replied:
All console logs removed

goofballLogic commented on script.js:
```diff
@@ -39,13 +39,26 @@ class App extends React.Component {
+		})
+		.then((response) => {
+			console.log(response);
+			if(response.data === 'Login succeeded'){this.setState({
```
We should probably base our logic on HTTP response status codes rather than text messages which might change.

dudeyds replied:
Logic now based on status response codes

dudeyds committed:
Update script.js

dudeyds committed:
time for toast

Added basic error handling in the UI, the error message handed by the server when there's an invalid username/ password combination now appears in an animated toast notification

goofballLogic commented on script.js:
```diff
@@ -39,13 +39,25 @@ class App extends React.Component {
+				} 
+			});
+			} else {
+				//Invalid password stuff here
```
```suggestion
				console.log("Not yet implemented: invalid password"); // TODO
```

dudeyds committed:
Update script.js

Remove a console log

dudeyds committed:
Merge pull request #7 from dudeyds/Toast-notification

Toast notifications added for error handling

dudeyds commented:
Bad password error handling, very basic form error handling and Toast notification UI added

goofballLogic commented:
Thanks @dudeyds I'll probably wait till Monday to get to this.

goofballLogic commented on script.js:
```diff
@@ -35,21 +35,37 @@ class App extends React.Component {
+			password: password
+		})
+		.then((response) => {
+			if(response.status === 201){this.setState({
```
At the moment, the server returns a 201, but if the server were in future to return a 200 (or 204 etc.) this piece of code would fail silently.

dudeyds replied:
catch all response now triggers a toast notification with the server's response displayed

goofballLogic commented on script.js:
```diff
@@ -35,21 +35,37 @@ class App extends React.Component {
+			});
+			}
+		}, (error) => {
+			if(error.response.status === 403) {this.setState( {
```
Similarly to the above, if the server were to throw an error (e.g. 500 error), or if a different type of error status were returned (e.g. 400 Bad Request), this code would fail silently and do nothing.

dudeyds replied:
method is now generalized so any error will be outputted as toast notification using the server's response as the text

goofballLogic reviewed (commented):
A few observations

dudeyds committed:
generalise toast triggers

Toast's now trigger on any response code other than the correct user/password response.

dudeyds committed:
Merge pull request #8 from dudeyds/Toast-notification

generalise toast triggers
